{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "WVV7j4WbJFQT"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import numpy as np"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "7-O6Z6sPEKAE"
      },
      "outputs": [],
      "source": [
        "#dff_1 = pd.read_csv('/content/drive/MyDrive/Susan_project/target_organism.csv')\n",
        "dff_1 = pd.read_csv(r'C:\\Users\\Igorr\\Documents\\ITMO5grade\\Project_with_Susan\\Made_code\\All_csv_files\\target_organism.csv')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1ZgMuPs1JyGI",
        "outputId": "935220c0-1835-4c64-8d22-30cd66a8cfe8"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "['Staphylococcus aureus',\n",
              " 'Streptococcus pneumoniae',\n",
              " 'Streptococcus pyogenes',\n",
              " 'Escherichia coli',\n",
              " 'Proteus',\n",
              " 'Salmonella',\n",
              " 'Klebsiella',\n",
              " 'Pseudomonas',\n",
              " 'Serratia',\n",
              " 'Enterobacter',\n",
              " 'Alcaligenes',\n",
              " 'Neisseria perflava',\n",
              " 'Staphylococcus',\n",
              " 'Bacillus',\n",
              " 'Streptococcus',\n",
              " 'Sarcina',\n",
              " 'Klebsiella pneumoniae',\n",
              " 'Salmonella enterica subsp. enterica serovar Typhimurium',\n",
              " 'Enterobacter cloacae',\n",
              " 'Haemophilus influenzae',\n",
              " 'Citrobacter gillenii',\n",
              " 'Shouchella clausii',\n",
              " 'Bacillus clausii',\n",
              " 'Aeromonas caviae',\n",
              " 'Citrobacter freundii',\n",
              " 'Enterobacteriaceae',\n",
              " \"Streptococcus sp. 'group A'\",\n",
              " \"Streptococcus sp. 'group B'\",\n",
              " 'Streptococcus viridans',\n",
              " 'Moraxella catarrhalis',\n",
              " 'Proteus mirabilis',\n",
              " 'Streptococcus pneumoniae R6',\n",
              " 'Streptococcus pneumoniae serotype 4 (strain ATCC BAA-334 / TIGR4)',\n",
              " 'Rattus norvegicus',\n",
              " 'Acinetobacter baumannii',\n",
              " 'Pseudomonas luteola',\n",
              " 'Pseudomonas aeruginosa',\n",
              " 'Salmonella enterica',\n",
              " 'Pseudomonas fluorescens',\n",
              " 'Providencia stuartii',\n",
              " 'Prescottella equi',\n",
              " 'Listeria monocytogenes',\n",
              " 'Corynebacterium amycolatum',\n",
              " 'Corynebacterium jeikeium',\n",
              " 'Corynebacterium pseudodiphtheriticum',\n",
              " 'Corynebacterium striatum',\n",
              " 'Corynebacterium urealyticum',\n",
              " 'Brachyspira pilosicoli',\n",
              " 'Stenotrophomonas maltophilia',\n",
              " 'Corynebacterium minutissimum',\n",
              " 'Corynebacterium xerosis',\n",
              " 'Pectobacterium atrosepticum',\n",
              " 'Elizabethkingia meningoseptica',\n",
              " 'Fluoribacter gormanii',\n",
              " 'Caulobacter vibrioides',\n",
              " 'Janthinobacterium lividum',\n",
              " 'Streptococcus constellatus',\n",
              " 'Streptococcus oralis',\n",
              " 'Streptococcus intermedius',\n",
              " 'Streptococcus mitis',\n",
              " 'Streptococcus agalactiae',\n",
              " 'Burkholderia cenocepacia',\n",
              " 'Acinetobacter genomosp. 3',\n",
              " 'Acinetobacter pittii',\n",
              " 'Neisseria meningitidis',\n",
              " 'Salmonella enterica subsp. enterica serovar Westhampton',\n",
              " 'Campylobacter jejuni',\n",
              " 'Campylobacter coli',\n",
              " 'Chryseobacterium indologenes',\n",
              " 'Flavobacterium meningosepticum',\n",
              " 'Yersinia enterocolitica',\n",
              " 'Staphylococcus epidermidis',\n",
              " 'Bacillus subtilis',\n",
              " 'Bacillus cereus',\n",
              " 'Micrococcus luteus',\n",
              " 'Bacillus spizizenii',\n",
              " 'Sus scrofa',\n",
              " 'Aeromonas hydrophila',\n",
              " 'Staphylococcus hominis',\n",
              " 'Enterococcus faecalis',\n",
              " 'Enterococcus faecium',\n",
              " 'Severe acute respiratory syndrome coronavirus 2',\n",
              " 'Staphylococcus lugdunensis',\n",
              " 'Staphylococcus saprophyticus',\n",
              " 'Serratia marcescens',\n",
              " 'Klebsiella aerogenes',\n",
              " 'Enterococcus sp.',\n",
              " 'Giardia intestinalis',\n",
              " 'Shigella sonnei',\n",
              " 'Klebsiella oxytoca',\n",
              " 'Trypanosoma brucei',\n",
              " 'Photinus pyralis',\n",
              " 'Salmonella enterica subsp. enterica serovar Paratyphi A',\n",
              " 'Shigella flexneri',\n",
              " 'Proteus vulgaris',\n",
              " 'Neisseria gonorrhoeae',\n",
              " 'Bacteroides fragilis',\n",
              " 'Fusobacterium varium',\n",
              " 'Clostridium perfringens',\n",
              " 'Canis lupus familiaris',\n",
              " 'Bacillus thuringiensis',\n",
              " 'Escherichia coli K-12',\n",
              " 'Enterococcus',\n",
              " 'Candida parapsilosis',\n",
              " 'Burkholderia cepacia',\n",
              " 'Nocardia cyriacigeorgica',\n",
              " 'Nocardia farcinica',\n",
              " 'Nocardia otitidiscaviarum',\n",
              " 'Nocardia nova',\n",
              " 'Nocardia transvalensis',\n",
              " 'Nocardia carnea',\n",
              " 'Nocardia brasiliensis',\n",
              " 'Nocardia sp.',\n",
              " 'Mycobacterium tuberculosis',\n",
              " 'Human immunodeficiency virus 1',\n",
              " 'Mycoplasmoides pneumoniae',\n",
              " 'Peptoniphilus asaccharolyticus',\n",
              " 'Peptostreptococcus anaerobius',\n",
              " 'Finegoldia magna',\n",
              " 'Cutibacterium acnes',\n",
              " 'Streptococcus pasteurianus',\n",
              " 'Streptococcus equinus',\n",
              " 'Acinetobacter',\n",
              " 'Metamycoplasma hominis',\n",
              " 'Ureaplasma',\n",
              " 'Plasmodium falciparum',\n",
              " 'Corynebacterium sp.',\n",
              " 'Mycolicibacterium smegmatis',\n",
              " 'Electrophorus electricus',\n",
              " 'Equus caballus',\n",
              " 'Leishmania donovani',\n",
              " 'Trypanosoma cruzi',\n",
              " 'Trypanosoma brucei rhodesiense',\n",
              " 'Klebsiella pneumoniae subsp. pneumoniae',\n",
              " 'Brugia malayi',\n",
              " 'Enterobacter sp.',\n",
              " '[Haemophilus] ducreyi',\n",
              " 'Bacillus anthracis',\n",
              " 'Legionella pneumophila',\n",
              " 'Shigella boydii',\n",
              " 'Shigella dysenteriae',\n",
              " 'Leptospira interrogans serovar Portlandvere',\n",
              " 'Chlamydia trachomatis',\n",
              " 'Ehrlichia canis',\n",
              " 'Mycoplasma putrefaciens',\n",
              " 'Plasmodium berghei',\n",
              " 'Dialister pneumosintes',\n",
              " 'Dialister micraerophilus',\n",
              " 'Dialister propionicifaciens',\n",
              " 'Dialister invisus',\n",
              " 'Enterococcus avium',\n",
              " 'Artemia salina',\n",
              " 'Chlamydia pneumoniae',\n",
              " 'Yersinia pestis',\n",
              " 'Brucella abortus',\n",
              " 'Brucella suis',\n",
              " 'Burkholderia pseudomallei',\n",
              " 'Mycobacteroides abscessus',\n",
              " 'Mycobacteroides chelonae',\n",
              " 'Mycolicibacterium peregrinum',\n",
              " 'Mycolicibacterium fortuitum',\n",
              " 'Mycolicibacterium mageritense',\n",
              " 'Saccharomyces cerevisiae',\n",
              " 'Vibrio sp.',\n",
              " 'Vibrio vulnificus',\n",
              " 'Vibrio fluvialis',\n",
              " 'Vibrio alginolyticus',\n",
              " 'Vibrio parahaemolyticus',\n",
              " 'Vibrio cholerae',\n",
              " 'Plesiomonas shigelloides',\n",
              " 'Aeromonas sobria',\n",
              " 'Yersinia pseudotuberculosis',\n",
              " 'Salmonella enterica subsp. enterica serovar Typhi',\n",
              " 'Salmonella enterica subsp. enterica serovar Enteritidis',\n",
              " 'Issatchenkia orientalis',\n",
              " 'Coxiella burnetii',\n",
              " 'Cavia porcellus',\n",
              " 'Oryctolagus cuniculus',\n",
              " 'Bos taurus',\n",
              " 'Plasmodium yoelii',\n",
              " 'Plasmodium vivax',\n",
              " 'Allium sativum',\n",
              " 'Salmonella enterica subsp. enterica serovar Abony',\n",
              " 'Streptococcus gordonii',\n",
              " 'dengue virus type 2',\n",
              " 'Litomosoides sigmodontis',\n",
              " 'Naegleria gruberi',\n",
              " 'Aspergillus fumigatus',\n",
              " 'Aspergillus terreus',\n",
              " 'Aspergillus flavus',\n",
              " 'Nakaseomyces glabratus',\n",
              " 'Absidia',\n",
              " 'Cunninghamella',\n",
              " 'Mucor',\n",
              " 'Rhizomucor',\n",
              " 'Rhizopus',\n",
              " 'Syncephalastrum',\n",
              " 'Wickerhamomyces anomalus',\n",
              " 'Candida albicans',\n",
              " 'Candida dubliniensis',\n",
              " 'Pichia kudriavzevii',\n",
              " 'Candida tropicalis',\n",
              " 'Aspergillus niger',\n",
              " 'Rhizopus arrhizus',\n",
              " 'Rhizopus microsporus var. rhizopodiformis',\n",
              " 'Rhizopus microsporus var. microsporus',\n",
              " 'Mucor circinelloides',\n",
              " 'Lichtheimia corymbifera',\n",
              " 'Apophysomyces elegans',\n",
              " 'Colletotrichum graminicola',\n",
              " 'Trichophyton mentagrophytes',\n",
              " 'Clavispora lusitaniae',\n",
              " 'Candida orthopsilosis',\n",
              " 'Candida metapsilosis',\n",
              " 'Neosartorya fumigata (Aspergillus fumigatus)',\n",
              " 'Aspergillus nidulans',\n",
              " 'Cryptococcus neoformans',\n",
              " 'Rhodotorula mucilaginosa',\n",
              " 'Trichosporon',\n",
              " 'Zygosaccharomyces',\n",
              " 'Kodamaea ohmeri',\n",
              " 'Cyberlindnera saturnus',\n",
              " 'Geotrichum',\n",
              " 'Kluyveromyces lactis',\n",
              " 'Pichia norvegensis',\n",
              " 'Nakaseomyces nivariensis',\n",
              " 'Pichia fermentans',\n",
              " 'Candida africana',\n",
              " '[Candida] intermedia',\n",
              " '[Candida] inconspicua',\n",
              " 'Cyberlindnera jadinii',\n",
              " 'Diutina rugosa',\n",
              " 'Debaryomyces hansenii',\n",
              " 'Yarrowia lipolytica',\n",
              " 'Meyerozyma caribbica',\n",
              " '[Candida] haemuloni',\n",
              " 'Kluyveromyces marxianus',\n",
              " 'Meyerozyma guilliermondii',\n",
              " 'Fusarium sp.',\n",
              " 'Fonsecaea pedrosoi',\n",
              " 'Fonsecaea monophora',\n",
              " 'Candida',\n",
              " 'Cryptococcus gattii',\n",
              " 'Trichosporon inkin',\n",
              " 'Trichosporon asahii',\n",
              " 'Geotrichum candidum',\n",
              " 'Mucor hiemalis',\n",
              " 'Lomentospora prolificans',\n",
              " 'Scedosporium apiospermum',\n",
              " 'Talaromyces marneffei',\n",
              " 'Penicillium sp.',\n",
              " 'Fusarium solani',\n",
              " 'Fusarium oxysporum',\n",
              " 'Sporothrix schenckii',\n",
              " 'Curvularia lunata',\n",
              " 'Curvularia australiensis',\n",
              " 'Trichophyton rubrum',\n",
              " 'Paecilomyces variotii',\n",
              " '[Candida] zeylanoides',\n",
              " 'Magnusiomyces capitatus',\n",
              " 'Cryptococcus neoformans var. neoformans',\n",
              " 'Naganishia albida',\n",
              " 'Rhodotorula glutinis',\n",
              " 'Papiliotrema laurentii',\n",
              " 'Aspergillus sydowii',\n",
              " 'Aspergillus versicolor',\n",
              " 'Aspergillus ochraceus',\n",
              " 'Aspergillus',\n",
              " 'Penicillium',\n",
              " 'Fusarium proliferatum',\n",
              " 'Fusarium verticillioides',\n",
              " 'Scopulariopsis brevicaulis',\n",
              " 'Trichoderma',\n",
              " 'Fusarium equiseti',\n",
              " 'Beauveria bassiana',\n",
              " 'Alternaria alternata',\n",
              " 'Exophiala dermatitidis',\n",
              " 'Aureobasidium pullulans',\n",
              " 'Cladosporium',\n",
              " 'Cunninghamella bertholletiae',\n",
              " 'Aspergillus tubingensis',\n",
              " 'Candida glabrata',\n",
              " 'Aspergillus brasiliensis',\n",
              " '[Candida] auris',\n",
              " 'Staphylococcus sp.',\n",
              " 'Streptococcus sp.',\n",
              " 'Staphylococcus haemolyticus',\n",
              " 'Mycobacterium avium',\n",
              " 'Staphylococcus capitis',\n",
              " 'Streptococcus dysgalactiae',\n",
              " 'Bacteria',\n",
              " 'Listeria innocua',\n",
              " 'Wolbachia',\n",
              " 'Pasteurella multocida',\n",
              " 'Priestia megaterium',\n",
              " 'Providencia rettgeri',\n",
              " 'Salmonella enterica subsp. enterica serovar Gallinarum',\n",
              " 'Streptococcus suis',\n",
              " 'Mycobacterium leprae',\n",
              " 'Vibrio splendidus',\n",
              " 'Vibrio anguillarum',\n",
              " 'Salmonella enterica subsp. enterica',\n",
              " 'Mycobacterium avium subsp. paratuberculosis',\n",
              " 'Mycobacterium tuberculosis (strain ATCC 25618 / H37Rv)',\n",
              " 'Influenza A virus',\n",
              " 'Lysinibacillus sphaericus',\n",
              " 'Acinetobacter lwoffii',\n",
              " 'Pseudomonas putida',\n",
              " 'Acinetobacter calcoaceticus',\n",
              " 'Bordetella pertussis',\n",
              " 'Lactococcus lactis',\n",
              " 'Morganella morganii',\n",
              " 'Plasmodium yoelii yoelii',\n",
              " 'Vibrio mimicus',\n",
              " 'Edwardsiella tarda',\n",
              " 'Cricetulus griseus',\n",
              " 'Bacteroides thetaiotaomicron',\n",
              " 'Prevotella melaninogenica',\n",
              " 'Bacteroides ovatus',\n",
              " 'Parabacteroides distasonis',\n",
              " 'Phocaeicola vulgatus',\n",
              " 'Fusobacterium nucleatum',\n",
              " 'Fusobacterium necrophorum',\n",
              " 'Thomasclavelia ramosa',\n",
              " 'Clostridium septicum',\n",
              " 'Clostridium butyricum',\n",
              " 'Clostridium sporogenes',\n",
              " 'Anaerococcus prevotii',\n",
              " 'Lycopus europaeus',\n",
              " 'Cercopithecidae',\n",
              " 'Bordetella bronchiseptica',\n",
              " 'Actinobacillus pleuropneumoniae',\n",
              " 'Trueperella pyogenes',\n",
              " 'Acinetobacter sp.',\n",
              " 'Bacillus subtilis (strain 168)',\n",
              " 'Macrophomina phaseolina',\n",
              " 'Trypanosoma brucei brucei',\n",
              " 'Mycolicibacterium phlei',\n",
              " 'Rhizoctonia solani',\n",
              " 'Ralstonia solanacearum',\n",
              " 'Erwinia',\n",
              " 'Xanthomonas axonopodis',\n",
              " 'Xanthomonas oryzae',\n",
              " 'Fusarium graminearum',\n",
              " 'Valsa mali',\n",
              " 'Escherichia coli (strain K12)',\n",
              " 'Streptococcus mutans',\n",
              " 'Mammaliicoccus sciuri',\n",
              " 'Mycobacterium kansasii',\n",
              " 'Clostridium tetani',\n",
              " 'Bacillus pumilus',\n",
              " 'Paracoccus denitrificans',\n",
              " 'Pseudomonas syringae',\n",
              " 'Corynebacterium',\n",
              " 'Aspergillus clavatus',\n",
              " 'Bacillus sp.',\n",
              " 'Bacillus amyloliquefaciens',\n",
              " 'Flavobacterium columnare',\n",
              " 'Enterococcus hirae',\n",
              " 'Proteus hauseri',\n",
              " 'Stachybotrys chartarum',\n",
              " 'Penicillium chrysogenum',\n",
              " 'Agaricus bisporus',\n",
              " 'Shigella',\n",
              " 'Hathewaya histolytica',\n",
              " 'Clostridioides difficile',\n",
              " 'Prevotella disiens',\n",
              " 'Hoylesella loescheii',\n",
              " 'Prevotella bivia',\n",
              " 'Fusobacterium',\n",
              " 'Veillonella parvula',\n",
              " 'Peptostreptococcus',\n",
              " 'Parvimonas micra',\n",
              " 'Mycobacterium intracellulare',\n",
              " 'Citrobacter koseri',\n",
              " 'Moraxella bovis',\n",
              " 'Helicobacter pylori',\n",
              " 'Aggregatibacter actinomycetemcomitans',\n",
              " 'Acinetobacter calcoaceticus subsp. anitratus',\n",
              " 'Alcaligenes faecalis',\n",
              " 'Proteus inconstans',\n",
              " 'Streptococcus sanguinis',\n",
              " 'Human immunodeficiency virus type 1 group M subtype B (isolate HXB2)(HIV-1)',\n",
              " 'Mycobacterium',\n",
              " 'Citrobacter',\n",
              " 'Chlorocebus sabaeus',\n",
              " 'Mycolicibacterium vaccae',\n",
              " 'Mycolicibacterium aurum',\n",
              " 'Chromobacterium violaceum',\n",
              " 'Salmonella enterica subsp. enterica serovar Typhi str. CT18',\n",
              " 'Mycobacterium tuberculosis variant microti',\n",
              " 'Haemophilus parainfluenzae',\n",
              " 'Klebsiella sp.',\n",
              " 'Delftia acidovorans',\n",
              " 'Staphylococcus cohnii',\n",
              " 'Staphylococcus warneri',\n",
              " 'Streptomyces griseus',\n",
              " 'Mycobacterium tuberculosis variant bovis',\n",
              " 'Acinetobacter radioresistens',\n",
              " 'Streptococcus salivarius',\n",
              " 'Pasteurella sp.',\n",
              " 'Corynebacterium glutamicum',\n",
              " 'Mammaliicoccus lentus',\n",
              " 'Staphylococcus xylosus',\n",
              " 'Enterobacter hormaechei',\n",
              " 'Citrobacter amalonaticus',\n",
              " 'Burkholderia thailandensis',\n",
              " 'Bacillus licheniformis',\n",
              " 'Brevibacillus brevis',\n",
              " 'Serratia sp.',\n",
              " 'Salmonella enterica subsp. enterica serovar Paratyphi B',\n",
              " 'Salmonella sp.',\n",
              " 'Streptomyces albus',\n",
              " 'Micrococcus',\n",
              " 'Human alphaherpesvirus 1',\n",
              " 'Lederbergia lenta',\n",
              " 'Plesiomonas',\n",
              " 'Francisella tularensis subsp. tularensis SCHU S4',\n",
              " 'Raoultella planticola',\n",
              " 'Acinetobacter haemolyticus',\n",
              " 'Toxoplasma gondii',\n",
              " 'Macaca mulatta',\n",
              " 'Kocuria rhizophila',\n",
              " 'Trypanosoma brucei gambiense',\n",
              " 'Burkholderia multivorans',\n",
              " 'Prevotella intermedia',\n",
              " 'Francisella tularensis',\n",
              " 'Rhodococcus erythropolis',\n",
              " 'Brucella neotomae',\n",
              " 'Mycobacterium avium complex sp.',\n",
              " 'Sporobolomyces salmonicolor',\n",
              " 'Enterococcus casseliflavus',\n",
              " 'Staphylococcus carnosus',\n",
              " 'Porphyromonas gingivalis',\n",
              " 'Proteus sp.',\n",
              " 'Clostridium beijerinckii',\n",
              " 'Clostridium botulinum',\n",
              " 'Bacillus flexus',\n",
              " 'Pyricularia grisea',\n",
              " 'Globisporangium ultimum',\n",
              " 'Mycobacterium marinum',\n",
              " 'Mycobacterium gordonae',\n",
              " 'Mycobacterium tuberculosis variant bovis BCG',\n",
              " 'Papio hamadryas',\n",
              " 'Mycobacterium xenopi',\n",
              " 'Mycobacterium ulcerans',\n",
              " 'Mycolicibacterium diernhoferi',\n",
              " 'Mesocricetus auratus',\n",
              " 'Artemia',\n",
              " 'Shigella sp.',\n",
              " 'Arthrobacter sp.',\n",
              " 'Mycobacterium scrofulaceum',\n",
              " 'Mycobacterium szulgai',\n",
              " 'Mycobacterium smegmatis',\n",
              " 'Mycolicibacter terrae',\n",
              " 'Mycobacterium smegmatis (strain ATCC 700084 / mc(2)155)',\n",
              " 'Mycobacterium tuberculosis H37Rv',\n",
              " 'Simian virus 40',\n",
              " 'Plasmodium falciparum 3D7',\n",
              " 'Chenopodium album',\n",
              " 'Sida spinosa',\n",
              " 'Senna obtusifolia',\n",
              " 'Ipomoea',\n",
              " 'Sorghum halepense',\n",
              " 'Sorghum bicolor',\n",
              " 'Poa nemoralis',\n",
              " 'Ambrosia artemisiifolia',\n",
              " 'Nasturtium officinale',\n",
              " 'Datura stramonium',\n",
              " 'Agrobacterium tumefaciens',\n",
              " 'Pectobacterium carotovorum',\n",
              " 'Curtobacterium flaccumfaciens',\n",
              " 'Colletotrichum acutatum',\n",
              " 'Carassius auratus',\n",
              " 'Entamoeba histolytica',\n",
              " 'Influenza A virus (strain A/USSR/90/1977 H1N1)',\n",
              " 'Cricetinae',\n",
              " 'Chikungunya virus',\n",
              " 'Influenza A virus (A/Puerto Rico/8/1934(H1N1))',\n",
              " 'Pseudomonas aeruginosa (strain ATCC 15692 / DSM 22644 / CIP 104116 / JCM14847 / LMG 12228 / 1C / PRS 101 / PAO1)',\n",
              " 'Zika virus',\n",
              " 'Hepacivirus hominis',\n",
              " 'Bifidobacterium bifidum',\n",
              " 'Bifidobacterium breve',\n",
              " 'Bifidobacterium longum',\n",
              " 'Bifidobacterium thermophilum',\n",
              " 'Streptomyces coelicolor',\n",
              " 'Microbotryum violaceum',\n",
              " 'Dickeya chrysanthemi',\n",
              " 'Streptomyces alboniger',\n",
              " 'Streptomyces acidiscabies',\n",
              " 'Streptomyces scabiei',\n",
              " 'Enterococcus gallinarum',\n",
              " 'Fusobacterium sp.',\n",
              " 'Prevotella buccae',\n",
              " 'Prevotella corporis',\n",
              " 'Hoylesella oralis',\n",
              " 'Actinomyces sp.',\n",
              " 'Streptococcus pyogenes serotype M1',\n",
              " 'Peptostreptococcus sp.',\n",
              " 'Scytonema hofmannii',\n",
              " 'Paenibacillus polymyxa',\n",
              " 'Bacteroides uniformis',\n",
              " 'Bacteroides caccae',\n",
              " 'Bacteroides stercoris',\n",
              " 'Parabacteroides merdae',\n",
              " '[Clostridium] innocuum',\n",
              " 'Clostridium paraputrificum',\n",
              " 'Clostridium tertium',\n",
              " 'Propionibacterium',\n",
              " 'Bacteroides sp.',\n",
              " 'Fusobacterium mortiferum',\n",
              " 'Porphyromonas asaccharolytica',\n",
              " 'Actinomyces israelii',\n",
              " 'Schaalia meyeri',\n",
              " 'Actinomyces naeslundii',\n",
              " 'Eggerthella lenta',\n",
              " 'Eubacterium sp.',\n",
              " 'Clostridium novyi',\n",
              " 'Clostridium cadaveris',\n",
              " 'Paeniclostridium sordellii',\n",
              " 'Paraclostridium bifermentans',\n",
              " 'Eubacterium limosum',\n",
              " 'Corynebacterium diphtheriae',\n",
              " 'Streptococcus equi subsp. zooepidemicus',\n",
              " 'Balamuthia mandrillaris',\n",
              " 'Chlamydia psittaci 6BC',\n",
              " 'Porphyromonas levii',\n",
              " 'Anaerococcus tetradius',\n",
              " 'Enterocloster clostridioformis',\n",
              " 'Prevotella nigrescens',\n",
              " 'Prevotella denticola',\n",
              " 'Corynebacterium aurimucosum',\n",
              " 'Corynebacterium accolens',\n",
              " 'Corynebacterium simulans',\n",
              " 'Streptococcus dysgalactiae subsp. equisimilis',\n",
              " 'Porphyromonas somerae',\n",
              " 'Anaerocolumna aminovalerica',\n",
              " 'Clostridium malenominatum',\n",
              " 'Lacrimispora sphenoides',\n",
              " 'Anaerococcus vaginalis',\n",
              " 'Veillonella',\n",
              " 'Collinsella aerofaciens',\n",
              " 'Lacticaseibacillus casei',\n",
              " 'Lacticaseibacillus rhamnosus',\n",
              " 'Lactobacillus acidophilus',\n",
              " 'Limosilactobacillus fermentum',\n",
              " 'Lactiplantibacillus plantarum',\n",
              " 'Limosilactobacillus reuteri',\n",
              " 'Capnocytophaga ochracea',\n",
              " 'Staphylococcus intermedius',\n",
              " 'Ruminococcus sp.',\n",
              " 'Plasmodium',\n",
              " 'Fannyhessea vaginae',\n",
              " 'Gardnerella vaginalis',\n",
              " 'Mobiluncus curtisii subsp. curtisii',\n",
              " 'Peptococcus niger',\n",
              " 'Treponema denticola',\n",
              " 'Pseudomonas aeruginosa (strain ATCC 15692 / PAO1 / 1C / PRS 101 / LMG12228)',\n",
              " 'Actinomyces viscosus',\n",
              " 'Lactobacillus jensenii',\n",
              " 'Lactobacillus sp.',\n",
              " 'Clostridium acetobutylicum',\n",
              " 'Oncorhynchus gorbuscha',\n",
              " 'Gerbillinae',\n",
              " 'Enterococcus durans',\n",
              " 'Staphylococcus simulans',\n",
              " 'Providencia sp.',\n",
              " 'Callithrix jacchus',\n",
              " 'Macaca fascicularis',\n",
              " 'Middle East respiratory syndrome-related coronavirus (isolate UnitedKingdom/H123990006/2012) (Betacoronavirus England 1) (Humancoronavirus EMC)',\n",
              " 'Torpedo californica',\n",
              " 'Human betaherpesvirus 5',\n",
              " 'Scheffersomyces shehatae',\n",
              " 'Scedosporium boydii',\n",
              " 'Lemna minor',\n",
              " 'Agrostis stolonifera var. palustris',\n",
              " 'Ustilago maydis',\n",
              " 'Phytophthora infestans',\n",
              " 'Pyricularia oryzae',\n",
              " 'Zymoseptoria tritici',\n",
              " 'Parastagonospora nodorum',\n",
              " 'Puccinia recondita',\n",
              " 'Blumeria graminis f. sp. tritici',\n",
              " 'Prevotella',\n",
              " 'Mycoplasmoides gallisepticum',\n",
              " 'Eikenella corrodens',\n",
              " 'Pseudomonas sp.',\n",
              " 'Cronobacter sakazakii',\n",
              " 'Staphylococcus chromogenes',\n",
              " 'Pluralibacter gergoviae',\n",
              " 'Caenorhabditis elegans',\n",
              " 'Serratia fonticola',\n",
              " 'Pseudomonas stutzeri',\n",
              " 'Acinetobacter baylyi',\n",
              " 'Aeromonas allosaccharophila',\n",
              " 'Leuconostoc',\n",
              " 'Rhodococcus sp.',\n",
              " 'Nocardia asteroides',\n",
              " 'Staphylococcus aureus subsp. aureus RN4220',\n",
              " 'Pseudomonas tolaasii',\n",
              " 'Micrococcus flavus',\n",
              " 'Pseudomonas savastanoi pv. savastanoi',\n",
              " 'Geobacillus stearothermophilus',\n",
              " 'Edwardsiella ictaluri',\n",
              " 'Kocuria',\n",
              " 'Bacillus mycoides',\n",
              " 'Xanthomonas campestris',\n",
              " '[Chlorella] fusca',\n",
              " 'Erwinia amylovora',\n",
              " 'Siminovitchia fortis',\n",
              " 'Meloidogyne',\n",
              " 'Streptomyces',\n",
              " 'Streptomyces thermovulgaris',\n",
              " 'Microsporum audouinii',\n",
              " 'Serratia liquefaciens',\n",
              " 'Pseudomonas oleovorans',\n",
              " 'Erysipelothrix rhusiopathiae',\n",
              " 'Aspergillus ruber',\n",
              " 'Mycotypha microspora',\n",
              " 'Pneumocystis carinii',\n",
              " 'Aeromonas sp.',\n",
              " 'Campylobacter fetus',\n",
              " 'Corynebacterium ammoniagenes',\n",
              " 'Penicillium griseofulvum',\n",
              " 'Trichoderma harzianum',\n",
              " 'Actinomadura sp. R39',\n",
              " 'Carnobacterium divergens',\n",
              " 'Penicillium expansum',\n",
              " 'Salmonella newport',\n",
              " 'Streptococcus pneumoniae (strain ATCC BAA-255 / R6)',\n",
              " 'Streptococcus sobrinus',\n",
              " 'Pseudomonas amygdali pv. lachrymans',\n",
              " 'Xanthomonas vesicatoria',\n",
              " 'Lactobacillus casei',\n",
              " 'Saccharomyces cerevisiae S288c',\n",
              " 'Plasmodium falciparum K1',\n",
              " 'Lactococcus lactis subsp. lactis (strain IL1403) (Streptococcuslactis)',\n",
              " 'Plasmodium berghei str. ANKA',\n",
              " 'Gallus gallus',\n",
              " 'Cryptosporidium parvum',\n",
              " 'Babesia gibsoni',\n",
              " 'Leishmania mexicana',\n",
              " 'Influenza B virus',\n",
              " 'Vesicular stomatitis virus',\n",
              " 'Coxsackievirus B4',\n",
              " 'Human respirovirus 3',\n",
              " 'Mammalian orthoreovirus 1',\n",
              " 'Sindbis virus',\n",
              " 'Punta Toro virus',\n",
              " 'Yellow fever virus',\n",
              " 'Respiratory syncytial virus',\n",
              " 'Leishmania major',\n",
              " 'Pneumocystis jirovecii',\n",
              " 'Hordeum vulgare',\n",
              " 'Ogataea angusta',\n",
              " 'Trypanosoma brucei brucei (strain 927/4 GUTat10.1)',\n",
              " 'Colletotrichum',\n",
              " 'Botrytis',\n",
              " 'Benjaminiella poitrasii',\n",
              " 'Lactobacillus johnsonii',\n",
              " 'Trichoderma viride',\n",
              " 'Brevibacterium epidermidis',\n",
              " 'Dermabacter hominis',\n",
              " 'Vibrio',\n",
              " 'Helicobacter pylori (strain ATCC 700392 / 26695) (Campylobacterpylori)',\n",
              " 'Photobacterium leiognathi',\n",
              " 'Nocardia canis',\n",
              " 'Xanthomonas euvesicatoria pv. sesami',\n",
              " 'Pseudomonas viridiflava',\n",
              " 'Paenibacillus larvae',\n",
              " 'Staphylococcus auricularis',\n",
              " 'Haemophilus',\n",
              " 'Trichomonas vaginalis',\n",
              " 'Leishmania infantum',\n",
              " 'Leishmania amazonensis',\n",
              " 'Stutzerimonas stutzeri',\n",
              " 'Mycolicibacterium flavescens',\n",
              " 'Escherichia coli O127:H6 (strain E2348/69 / EPEC)',\n",
              " 'Providencia alcalifaciens',\n",
              " 'Bacillus thermoproteolyticus',\n",
              " 'Vibrio proteolyticus',\n",
              " 'Human immunodeficiency virus 2',\n",
              " 'Acholeplasma laidlawii',\n",
              " 'Echinococcus equinus',\n",
              " 'Yamadazyma tenuis',\n",
              " 'Staphylococcus aureus (strain Mu50 / ATCC 700699)',\n",
              " 'Lactobacillus delbrueckii',\n",
              " 'Lacticaseibacillus paracasei',\n",
              " 'Phocaeicola dorei',\n",
              " 'Lactobacillus gasseri',\n",
              " 'Felis catus',\n",
              " 'Haemophilus influenzae (strain ATCC 51907 / DSM 11121 / KW20 / Rd)',\n",
              " 'Schistosoma mansoni',\n",
              " 'Trichophyton tonsurans',\n",
              " 'Microsporum',\n",
              " 'Trichophyton benhamiae',\n",
              " 'Vibrio harveyi',\n",
              " 'Escherichia coli (strain SMS-3-5 / SECEC)',\n",
              " 'Aspergillus sp.',\n",
              " 'Botrytis fabae',\n",
              " 'Candida maltosa',\n",
              " 'Nannizzia gypsea',\n",
              " 'Epidermophyton floccosum',\n",
              " 'Trichophyton interdigitale',\n",
              " 'Cladosporium cladosporioides',\n",
              " 'Cladosporium herbarum',\n",
              " 'Streptomyces chartreusis',\n",
              " 'Rhizomucor miehei',\n",
              " 'Alternaria solani',\n",
              " 'Xanthomonas oryzae pv. oryzae',\n",
              " 'Vibrio aestuarianus',\n",
              " 'Lasiodiplodia theobromae',\n",
              " 'Brevundimonas diminuta',\n",
              " 'Ochrobactrum',\n",
              " 'Pantoea dispersa',\n",
              " 'Coxsackievirus B1',\n",
              " 'Fusarium oxysporum f. sp. cubense',\n",
              " 'Pseudoalteromonas lipolytica',\n",
              " 'Pseudoalteromonas ulvae',\n",
              " 'Paracoccus sp.',\n",
              " 'Rhodococcus fascians',\n",
              " 'Pichia kluyveri',\n",
              " 'Sphingomonas paucimobilis',\n",
              " 'Trichoplusia ni',\n",
              " 'Setaria cervi',\n",
              " 'Dirofilaria immitis',\n",
              " 'Danio rerio',\n",
              " 'Neisseria',\n",
              " 'Arthroderma uncinatum',\n",
              " 'Nannizzia nana',\n",
              " 'Paraphyton cookei',\n",
              " 'Kocuria rosea',\n",
              " 'Saccharomyces',\n",
              " 'Trichophyton tonsurans var. sulfureum',\n",
              " 'Penicillium italicum',\n",
              " 'Syncephalastrum racemosum',\n",
              " 'Pseudomonas savastanoi pv. phaseolicola',\n",
              " 'Sinorhizobium meliloti',\n",
              " 'Heyndrickxia coagulans',\n",
              " 'Scopulariopsis',\n",
              " 'Cricetinae gen. sp.',\n",
              " 'Microsporum canis',\n",
              " 'Phialophora verrucosa',\n",
              " 'Saprolegnia',\n",
              " 'Histoplasma capsulatum',\n",
              " 'Hanseniaspora guilliermondii',\n",
              " '[Candida] sake',\n",
              " 'Plasmodium chabaudi',\n",
              " 'Aliivibrio fischeri',\n",
              " 'Rhodococcus corynebacterioides',\n",
              " 'Aspergillus parasiticus',\n",
              " 'Lipomyces lipofer',\n",
              " 'Neosartorya fumigata (strain ATCC MYA-4609 / Af293 / CBS 101355 / FGSCA1100) (Aspergillus fumigatus)',\n",
              " 'Candida albicans (strain WO-1) (Yeast)',\n",
              " 'Candida dubliniensis (strain CD36 / CBS 7987 / NCPF 3949 / NRRLY-17841) (Yeast)',\n",
              " 'Colletotrichum truncatum',\n",
              " 'Pythium aphanidermatum',\n",
              " 'Trichophyton longifusum',\n",
              " 'Aspergillus awamori',\n",
              " 'Monascus purpureus',\n",
              " 'Leishmania chagasi',\n",
              " 'Mucor racemosus',\n",
              " 'Mus sp.',\n",
              " 'Trichosporon beigelii',\n",
              " 'Trichophyton violaceum',\n",
              " 'Coccidioides immitis',\n",
              " 'Aspergillus oryzae',\n",
              " 'Fonsecaea compacta',\n",
              " '[Candida] palmioleophila',\n",
              " 'Apiotrichum loubieri',\n",
              " 'Cutaneotrichosporon mucoides',\n",
              " 'Starmerella stellata',\n",
              " 'Hanseniaspora uvarum',\n",
              " 'Rhodotorula',\n",
              " 'Saccharomycopsis',\n",
              " 'Sporothrix',\n",
              " 'Sporothrix globosa',\n",
              " 'Sporothrix albicans',\n",
              " 'Penicillium citrinum',\n",
              " 'Curvularia',\n",
              " 'Malassezia furfur',\n",
              " 'Fusarium',\n",
              " 'Exophiala spinifera',\n",
              " 'Candida albicans (strain SC5314 / ATCC MYA-2876) (Yeast)',\n",
              " 'Botrytis cinerea',\n",
              " '[Candida] aaseri',\n",
              " 'Mucor fuscus',\n",
              " 'Human immunodeficiency virus',\n",
              " 'Goffeauzyma gastrica',\n",
              " 'Chrysosporium keratinophilum',\n",
              " 'Pichia sporocuriosa',\n",
              " 'Trypanosoma cruzi (strain CL Brener)',\n",
              " \"Saccharomyces cerevisiae (strain ATCC 204508 / S288c) (Baker's yeast)\",\n",
              " 'Malassezia pachydermatis',\n",
              " 'Arthrodermataceae',\n",
              " 'Trichophyton verrucosum',\n",
              " 'Leishmania braziliensis',\n",
              " 'Leishmania panamensis',\n",
              " \"Trichophyton rubrum (strain ATCC MYA-4607 / CBS 118892) (Athlete'sfoot fungus)\",\n",
              " 'Paracoccidioides brasiliensis',\n",
              " 'Kregervanrija fluxuum',\n",
              " 'Cryptococcus gattii species complex',\n",
              " 'Fungi',\n",
              " 'Mycobacterium simiae',\n",
              " 'Mycobacterium malmoense',\n",
              " 'Chlamydia',\n",
              " 'Naegleria fowleri',\n",
              " 'Zaire ebolavirus',\n",
              " 'Pseudomonas aeruginosa PAO1',\n",
              " 'Yersinia sp.',\n",
              " 'Aeromonas veronii',\n",
              " 'Tsukamurella pulmonis',\n",
              " 'Spinacia oleracea']"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "clean_bact = dff_1['target_organism'].tolist()\n",
        "clean_bact"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 191
        },
        "id": "Y7PDj177arRk",
        "outputId": "486404c5-320a-41be-d1ff-f31f6b3b70ea"
      },
      "outputs": [],
      "source": [
        "# # Извлекаем первые 10 бактерий\n",
        "# first_10_bacteria = clean_bact[:10]\n",
        "\n",
        "# # Выводим результат\n",
        "# display(first_10_bacteria)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "_aRaS3YrASlh"
      },
      "outputs": [],
      "source": [
        "\n",
        "\n",
        "# # Список элементов для удаления\n",
        "# elements_to_remove = [\n",
        "#     \"Streptococcus sp. 'group B'\",\n",
        "#     \"Streptococcus sp. 'group A'\",\n",
        "#     \"Salmonella enterica subsp. enterica serovar Typhimurium\",'Streptococcus pneumoniae R6',\n",
        "#  'Streptococcus pneumoniae serotype 4 (strain ATCC BAA-334 / TIGR4)', 'Salmonella enterica subsp. enterica serovar Westhampton', 'Severe acute respiratory syndrome coronavirus 2','Salmonella enterica subsp. enterica serovar Paratyphi A','Salmonella enterica subsp. enterica serovar Abony', 'Salmonella enterica subsp. enterica serovar Typhi',\n",
        "#  'Salmonella enterica subsp. enterica serovar Enteritidis'\n",
        "# ]\n",
        "\n",
        "# # Удаляем элементы\n",
        "# my_list = [item for item in clean_bact if item not in elements_to_remove]\n",
        "\n",
        "# # Результат\n",
        "# my_list\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "yq45BHhxEJ9o"
      },
      "outputs": [],
      "source": [
        "# import time\n",
        "# ID_list = []\n",
        "# from Bio import Entrez\n",
        "# Entrez.email = \"iroslov.01@gmail.com\"     # Always tell NCBI who you are\n",
        "# for bact in first_10_bacteria:\n",
        "#     handle = Entrez.esearch(db=\"Taxonomy\", term=bact)\n",
        "#     record = Entrez.read(handle)\n",
        "#     ID_list.append(record)\n",
        "#     time.sleep(1)  # add a 1-second delay between each request\n",
        "#     print(record)\n",
        "# print(ID_list)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "hsGKdlBiEJ7E"
      },
      "outputs": [],
      "source": [
        "# df =pd.DataFrame(ID_list)\n",
        "# df['name'] = first_10_bacteria\n",
        "# df"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3bkL-QGsgdoX"
      },
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "FELTcDcJ8032"
      },
      "outputs": [],
      "source": [
        "# df.to_csv('/content/drive/MyDrive/Susan_project/taxonomy_descriptor.csv', index=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 339
        },
        "id": "lovAhG2XVmvR",
        "outputId": "98b6a4d0-a13b-456c-824b-f528e470031d"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Count</th>\n",
              "      <th>RetMax</th>\n",
              "      <th>RetStart</th>\n",
              "      <th>IdList</th>\n",
              "      <th>TranslationSet</th>\n",
              "      <th>TranslationStack</th>\n",
              "      <th>QueryTranslation</th>\n",
              "      <th>ErrorList</th>\n",
              "      <th>WarningList</th>\n",
              "      <th>name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1280']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'staphylococcus aureus[All Names]', ...</td>\n",
              "      <td>staphylococcus aureus[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Staphylococcus aureus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1313']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'streptococcus pneumoniae[All Names]...</td>\n",
              "      <td>streptococcus pneumoniae[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Streptococcus pneumoniae</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1314']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'streptococcus pyogenes[All Names]',...</td>\n",
              "      <td>streptococcus pyogenes[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Streptococcus pyogenes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['562']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'escherichia coli[All Names]', 'Fiel...</td>\n",
              "      <td>escherichia coli[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Escherichia coli</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "      <td>['210425', '583']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Proteus[All Names]', 'Field': 'All ...</td>\n",
              "      <td>Proteus[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Proteus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>809</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['208964']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Pseudomonas aeruginosa PAO1[All Nam...</td>\n",
              "      <td>Pseudomonas aeruginosa PAO1[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Pseudomonas aeruginosa PAO1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>810</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>[]</td>\n",
              "      <td>[]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>(Yersinia sp.[All Names])</td>\n",
              "      <td>{'FieldNotFound': [], 'PhraseNotFound': ['Yers...</td>\n",
              "      <td>{'PhraseIgnored': [], 'OutputMessage': ['No it...</td>\n",
              "      <td>Yersinia sp.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>811</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['654']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Aeromonas veronii[All Names]', 'Fie...</td>\n",
              "      <td>Aeromonas veronii[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Aeromonas veronii</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>812</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['47312']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Tsukamurella pulmonis[All Names]', ...</td>\n",
              "      <td>Tsukamurella pulmonis[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Tsukamurella pulmonis</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>813</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['3562']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'spinacia oleracea[All Names]', 'Fie...</td>\n",
              "      <td>spinacia oleracea[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Spinacia oleracea</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>814 rows × 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     Count  RetMax  RetStart             IdList TranslationSet  \\\n",
              "0        1       1         0           ['1280']             []   \n",
              "1        1       1         0           ['1313']             []   \n",
              "2        1       1         0           ['1314']             []   \n",
              "3        1       1         0            ['562']             []   \n",
              "4        2       2         0  ['210425', '583']             []   \n",
              "..     ...     ...       ...                ...            ...   \n",
              "809      1       1         0         ['208964']             []   \n",
              "810      0       0         0                 []             []   \n",
              "811      1       1         0            ['654']             []   \n",
              "812      1       1         0          ['47312']             []   \n",
              "813      1       1         0           ['3562']             []   \n",
              "\n",
              "                                      TranslationStack  \\\n",
              "0    [{'Term': 'staphylococcus aureus[All Names]', ...   \n",
              "1    [{'Term': 'streptococcus pneumoniae[All Names]...   \n",
              "2    [{'Term': 'streptococcus pyogenes[All Names]',...   \n",
              "3    [{'Term': 'escherichia coli[All Names]', 'Fiel...   \n",
              "4    [{'Term': 'Proteus[All Names]', 'Field': 'All ...   \n",
              "..                                                 ...   \n",
              "809  [{'Term': 'Pseudomonas aeruginosa PAO1[All Nam...   \n",
              "810                                                NaN   \n",
              "811  [{'Term': 'Aeromonas veronii[All Names]', 'Fie...   \n",
              "812  [{'Term': 'Tsukamurella pulmonis[All Names]', ...   \n",
              "813  [{'Term': 'spinacia oleracea[All Names]', 'Fie...   \n",
              "\n",
              "                           QueryTranslation  \\\n",
              "0          staphylococcus aureus[All Names]   \n",
              "1       streptococcus pneumoniae[All Names]   \n",
              "2         streptococcus pyogenes[All Names]   \n",
              "3               escherichia coli[All Names]   \n",
              "4                        Proteus[All Names]   \n",
              "..                                      ...   \n",
              "809  Pseudomonas aeruginosa PAO1[All Names]   \n",
              "810               (Yersinia sp.[All Names])   \n",
              "811            Aeromonas veronii[All Names]   \n",
              "812        Tsukamurella pulmonis[All Names]   \n",
              "813            spinacia oleracea[All Names]   \n",
              "\n",
              "                                             ErrorList  \\\n",
              "0                                                  NaN   \n",
              "1                                                  NaN   \n",
              "2                                                  NaN   \n",
              "3                                                  NaN   \n",
              "4                                                  NaN   \n",
              "..                                                 ...   \n",
              "809                                                NaN   \n",
              "810  {'FieldNotFound': [], 'PhraseNotFound': ['Yers...   \n",
              "811                                                NaN   \n",
              "812                                                NaN   \n",
              "813                                                NaN   \n",
              "\n",
              "                                           WarningList  \\\n",
              "0                                                  NaN   \n",
              "1                                                  NaN   \n",
              "2                                                  NaN   \n",
              "3                                                  NaN   \n",
              "4                                                  NaN   \n",
              "..                                                 ...   \n",
              "809                                                NaN   \n",
              "810  {'PhraseIgnored': [], 'OutputMessage': ['No it...   \n",
              "811                                                NaN   \n",
              "812                                                NaN   \n",
              "813                                                NaN   \n",
              "\n",
              "                            name  \n",
              "0          Staphylococcus aureus  \n",
              "1       Streptococcus pneumoniae  \n",
              "2         Streptococcus pyogenes  \n",
              "3               Escherichia coli  \n",
              "4                        Proteus  \n",
              "..                           ...  \n",
              "809  Pseudomonas aeruginosa PAO1  \n",
              "810                 Yersinia sp.  \n",
              "811            Aeromonas veronii  \n",
              "812        Tsukamurella pulmonis  \n",
              "813            Spinacia oleracea  \n",
              "\n",
              "[814 rows x 10 columns]"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = pd.read_csv(r'C:\\Users\\Igorr\\Documents\\ITMO5grade\\Project_with_Susan\\Made_code\\All_csv_files\\taxonomy_descriptor.csv')\n",
        "df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "id": "QIIayGkt5JYI"
      },
      "outputs": [],
      "source": [
        "# df.to_excel('/content/drive/MyDrive/Susan_project/taxonomy_descriptor.xlsx')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "g7An5JgUEJ4t"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Count</th>\n",
              "      <th>RetMax</th>\n",
              "      <th>RetStart</th>\n",
              "      <th>IdList</th>\n",
              "      <th>TranslationSet</th>\n",
              "      <th>TranslationStack</th>\n",
              "      <th>QueryTranslation</th>\n",
              "      <th>ErrorList</th>\n",
              "      <th>WarningList</th>\n",
              "      <th>name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1280']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'staphylococcus aureus[All Names]', ...</td>\n",
              "      <td>staphylococcus aureus[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Staphylococcus aureus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1313']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'streptococcus pneumoniae[All Names]...</td>\n",
              "      <td>streptococcus pneumoniae[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Streptococcus pneumoniae</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['1314']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'streptococcus pyogenes[All Names]',...</td>\n",
              "      <td>streptococcus pyogenes[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Streptococcus pyogenes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['562']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'escherichia coli[All Names]', 'Fiel...</td>\n",
              "      <td>escherichia coli[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Escherichia coli</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "      <td>['210425', '583']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Proteus[All Names]', 'Field': 'All ...</td>\n",
              "      <td>Proteus[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Proteus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>808</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['186538']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Zaire ebolavirus[All Names]', 'Fiel...</td>\n",
              "      <td>Zaire ebolavirus[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Zaire ebolavirus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>809</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['208964']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Pseudomonas aeruginosa PAO1[All Nam...</td>\n",
              "      <td>Pseudomonas aeruginosa PAO1[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Pseudomonas aeruginosa PAO1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>811</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['654']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Aeromonas veronii[All Names]', 'Fie...</td>\n",
              "      <td>Aeromonas veronii[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Aeromonas veronii</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>812</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['47312']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'Tsukamurella pulmonis[All Names]', ...</td>\n",
              "      <td>Tsukamurella pulmonis[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Tsukamurella pulmonis</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>813</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>['3562']</td>\n",
              "      <td>[]</td>\n",
              "      <td>[{'Term': 'spinacia oleracea[All Names]', 'Fie...</td>\n",
              "      <td>spinacia oleracea[All Names]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Spinacia oleracea</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>771 rows × 10 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "     Count  RetMax  RetStart             IdList TranslationSet  \\\n",
              "0        1       1         0           ['1280']             []   \n",
              "1        1       1         0           ['1313']             []   \n",
              "2        1       1         0           ['1314']             []   \n",
              "3        1       1         0            ['562']             []   \n",
              "4        2       2         0  ['210425', '583']             []   \n",
              "..     ...     ...       ...                ...            ...   \n",
              "808      1       1         0         ['186538']             []   \n",
              "809      1       1         0         ['208964']             []   \n",
              "811      1       1         0            ['654']             []   \n",
              "812      1       1         0          ['47312']             []   \n",
              "813      1       1         0           ['3562']             []   \n",
              "\n",
              "                                      TranslationStack  \\\n",
              "0    [{'Term': 'staphylococcus aureus[All Names]', ...   \n",
              "1    [{'Term': 'streptococcus pneumoniae[All Names]...   \n",
              "2    [{'Term': 'streptococcus pyogenes[All Names]',...   \n",
              "3    [{'Term': 'escherichia coli[All Names]', 'Fiel...   \n",
              "4    [{'Term': 'Proteus[All Names]', 'Field': 'All ...   \n",
              "..                                                 ...   \n",
              "808  [{'Term': 'Zaire ebolavirus[All Names]', 'Fiel...   \n",
              "809  [{'Term': 'Pseudomonas aeruginosa PAO1[All Nam...   \n",
              "811  [{'Term': 'Aeromonas veronii[All Names]', 'Fie...   \n",
              "812  [{'Term': 'Tsukamurella pulmonis[All Names]', ...   \n",
              "813  [{'Term': 'spinacia oleracea[All Names]', 'Fie...   \n",
              "\n",
              "                           QueryTranslation ErrorList WarningList  \\\n",
              "0          staphylococcus aureus[All Names]       NaN         NaN   \n",
              "1       streptococcus pneumoniae[All Names]       NaN         NaN   \n",
              "2         streptococcus pyogenes[All Names]       NaN         NaN   \n",
              "3               escherichia coli[All Names]       NaN         NaN   \n",
              "4                        Proteus[All Names]       NaN         NaN   \n",
              "..                                      ...       ...         ...   \n",
              "808             Zaire ebolavirus[All Names]       NaN         NaN   \n",
              "809  Pseudomonas aeruginosa PAO1[All Names]       NaN         NaN   \n",
              "811            Aeromonas veronii[All Names]       NaN         NaN   \n",
              "812        Tsukamurella pulmonis[All Names]       NaN         NaN   \n",
              "813            spinacia oleracea[All Names]       NaN         NaN   \n",
              "\n",
              "                            name  \n",
              "0          Staphylococcus aureus  \n",
              "1       Streptococcus pneumoniae  \n",
              "2         Streptococcus pyogenes  \n",
              "3               Escherichia coli  \n",
              "4                        Proteus  \n",
              "..                           ...  \n",
              "808             Zaire ebolavirus  \n",
              "809  Pseudomonas aeruginosa PAO1  \n",
              "811            Aeromonas veronii  \n",
              "812        Tsukamurella pulmonis  \n",
              "813            Spinacia oleracea  \n",
              "\n",
              "[771 rows x 10 columns]"
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df.dropna(subset=['TranslationStack'], inplace=True)\n",
        "df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "FqrImvDbEJ2E"
      },
      "outputs": [],
      "source": [
        "dff = df[['IdList', 'name']]\n",
        "# dff.iloc[[10]]\n",
        "# dfr = dff.drop(dff.index[10]) # removing 10th row\n",
        "dfr = dff.reset_index(drop=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "id": "1XZ1wNchEJzu"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>IdList</th>\n",
              "      <th>name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>['1280']</td>\n",
              "      <td>Staphylococcus aureus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>['1313']</td>\n",
              "      <td>Streptococcus pneumoniae</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>['1314']</td>\n",
              "      <td>Streptococcus pyogenes</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>['562']</td>\n",
              "      <td>Escherichia coli</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>['210425', '583']</td>\n",
              "      <td>Proteus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>766</th>\n",
              "      <td>['186538']</td>\n",
              "      <td>Zaire ebolavirus</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>767</th>\n",
              "      <td>['208964']</td>\n",
              "      <td>Pseudomonas aeruginosa PAO1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>768</th>\n",
              "      <td>['654']</td>\n",
              "      <td>Aeromonas veronii</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>769</th>\n",
              "      <td>['47312']</td>\n",
              "      <td>Tsukamurella pulmonis</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>770</th>\n",
              "      <td>['3562']</td>\n",
              "      <td>Spinacia oleracea</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>771 rows × 2 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "                IdList                         name\n",
              "0             ['1280']        Staphylococcus aureus\n",
              "1             ['1313']     Streptococcus pneumoniae\n",
              "2             ['1314']       Streptococcus pyogenes\n",
              "3              ['562']             Escherichia coli\n",
              "4    ['210425', '583']                      Proteus\n",
              "..                 ...                          ...\n",
              "766         ['186538']             Zaire ebolavirus\n",
              "767         ['208964']  Pseudomonas aeruginosa PAO1\n",
              "768            ['654']            Aeromonas veronii\n",
              "769          ['47312']        Tsukamurella pulmonis\n",
              "770           ['3562']            Spinacia oleracea\n",
              "\n",
              "[771 rows x 2 columns]"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# dfr = dfr.drop([463,327,277])\n",
        "# dfr.reset_index(drop=True, inplace=True)\n",
        "dfr"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "id": "zF-Z_UR91EQ_"
      },
      "outputs": [],
      "source": [
        "import ast"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "WIrHNFrQ1FJK"
      },
      "outputs": [],
      "source": [
        "def convert_to_list(x):\n",
        "    if isinstance(x, str):\n",
        "        x = x.strip().strip(\"'\\\"\")  # Удаляем лишние пробелы и внешние кавычки\n",
        "        try:\n",
        "            x = ast.literal_eval(x)\n",
        "        except Exception as e:\n",
        "            print(f\"Ошибка при преобразовании в список на значении {x}: {e}\")\n",
        "            x = np.nan\n",
        "    return x"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "id": "8E1KPBI91JK_"
      },
      "outputs": [],
      "source": [
        "dfr['IdList'] = dfr['IdList'].apply(convert_to_list)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "He0EBe5Z1KH5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "IdList\n",
            "<class 'list'>    771\n",
            "Name: count, dtype: int64\n"
          ]
        }
      ],
      "source": [
        "print(dfr['IdList'].apply(type).value_counts())\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "id": "-H3QZF8P1NgA"
      },
      "outputs": [],
      "source": [
        "def remove_first_value(x):\n",
        "    if isinstance(x, (list, ListElement)):\n",
        "        if len(x) > 1:\n",
        "            return x[1]  # Возвращаем второй элемент списка\n",
        "        elif len(x) == 1:\n",
        "            return x[0]  # Если только один элемент, возвращаем его\n",
        "        else:\n",
        "            return np.nan\n",
        "    else:\n",
        "        return x  # Возвращаем значение как есть, если оно не список\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "GSzdOLHe2J-p"
      },
      "outputs": [
        {
          "ename": "ModuleNotFoundError",
          "evalue": "No module named 'Bio'",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
            "Cell \u001b[1;32mIn[15], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mtime\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Entrez\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mEntrez\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mParser\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ListElement\n\u001b[0;32m      4\u001b[0m Entrez\u001b[38;5;241m.\u001b[39memail \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miroslov.01@gmail.com\u001b[39m\u001b[38;5;124m\"\u001b[39m\n",
            "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'Bio'"
          ]
        }
      ],
      "source": [
        "import time\n",
        "from Bio import Entrez\n",
        "from Bio.Entrez.Parser import ListElement\n",
        "Entrez.email = \"iroslov.01@gmail.com\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "id": "KadlffrQ1Rif"
      },
      "outputs": [],
      "source": [
        "dfr['IdList'] = dfr['IdList'].apply(remove_first_value)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "id": "y4aNebKY1S0X"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "  IdList                      name\n",
            "0   1280     Staphylococcus aureus\n",
            "1   1313  Streptococcus pneumoniae\n",
            "2   1314    Streptococcus pyogenes\n",
            "3    562          Escherichia coli\n",
            "4    583                   Proteus\n",
            "5    590                Salmonella\n",
            "6    570                Klebsiella\n",
            "7    286               Pseudomonas\n",
            "8    613                  Serratia\n",
            "9    547              Enterobacter\n"
          ]
        }
      ],
      "source": [
        "print(dfr[['IdList', 'name']].head(10))\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "id": "WnSIbOSv1V9-"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Ошибка при обработке ID 235 на индексе 152: Remote end closed connection without response\n",
            "Ошибка при обработке ID 29461 на индексе 153: <urlopen error [WinError 10060] Попытка установить соединение была безуспешной, т.к. от другого компьютера за требуемое время не получен нужный отклик, или было разорвано уже установленное соединение из-за неверного отклика уже подключенного компьютера>\n",
            "Ошибка при обработке ID 5535 на индексе 256: HTTP Error 400: Bad Request\n",
            "Некорректный ID на индексе 277: nan\n",
            "Ошибка при обработке ID 1334 на индексе 283: Remote end closed connection without response\n",
            "Ошибка при обработке ID 59201 на индексе 295: <urlopen error [Errno 11001] getaddrinfo failed>\n",
            "Ошибка при обработке ID 1770 на индексе 296: <urlopen error [Errno 11001] getaddrinfo failed>\n",
            "Ошибка при обработке ID 11320 на индексе 297: <urlopen error [Errno 11001] getaddrinfo failed>\n",
            "Некорректный ID на индексе 327: nan\n",
            "Некорректный ID на индексе 463: nan\n",
            "Ошибка при обработке ID 11034 на индексе 625: HTTP Error 400: Bad Request\n",
            "Ошибка при обработке ID 5506 на индексе 745: HTTP Error 400: Bad Request\n",
            "0      Bacteria\n",
            "1      Bacteria\n",
            "2      Bacteria\n",
            "3      Bacteria\n",
            "4      Bacteria\n",
            "         ...   \n",
            "766         NaN\n",
            "767    Bacteria\n",
            "768    Bacteria\n",
            "769    Bacteria\n",
            "770         NaN\n",
            "Name: Kingdom, Length: 771, dtype: object\n"
          ]
        }
      ],
      "source": [
        "# Создаем пустой список для хранения уровня таксономической классификации \"Kingdom\"\n",
        "kingdom = []\n",
        "\n",
        "# Создаем пустые DataFrame для бактерий и грибов\n",
        "bacteria_df = pd.DataFrame()\n",
        "fungi_df = pd.DataFrame()\n",
        "\n",
        "# Итерация по DataFrame\n",
        "for i in range(len(dfr)):\n",
        "    id_value = dfr['IdList'].iloc[i]\n",
        "\n",
        "    # Проверяем, является ли ID допустимым\n",
        "    if pd.isna(id_value) or not isinstance(id_value, (str, int)):\n",
        "        print(f\"Некорректный ID на индексе {i}: {id_value}\")\n",
        "        kingdom.append(np.nan)\n",
        "        continue\n",
        "\n",
        "    id_value = str(id_value)  # Преобразуем ID в строку\n",
        "\n",
        "    try:\n",
        "        handle = Entrez.efetch(db=\"Taxonomy\", id=id_value, retmode=\"xml\")\n",
        "        records = Entrez.read(handle)\n",
        "        handle.close()\n",
        "\n",
        "        # Получаем таксономическую линию\n",
        "        tax = records[0][\"Lineage\"].split(\"; \")\n",
        "        time.sleep(0.2)\n",
        "\n",
        "        # Проверяем, являются ли это грибами или бактериями\n",
        "        if any(\"Fungi\" in tax_elem for tax_elem in tax):\n",
        "            # Добавляем уровень \"Kingdom\" в список\n",
        "            kingdom.append(tax[3] if len(tax) > 3 else np.nan)\n",
        "            # Добавляем данные в DataFrame грибов\n",
        "            fungi_df = pd.concat([fungi_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "        elif any(\"Bacteria\" in tax_elem for tax_elem in tax):\n",
        "            # Добавляем уровень \"Kingdom\" в список\n",
        "            kingdom.append(tax[1] if len(tax) > 1 else np.nan)\n",
        "            # Добавляем данные в DataFrame бактерий\n",
        "            bacteria_df = pd.concat([bacteria_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "        else:\n",
        "            kingdom.append(np.nan)\n",
        "    except Exception as e:\n",
        "        print(f\"Ошибка при обработке ID {id_value} на индексе {i}: {e}\")\n",
        "        kingdom.append(np.nan)\n",
        "\n",
        "# Создаем новый столбец в DataFrame для уровня классификации \"Kingdom\"\n",
        "dfr[\"Kingdom\"] = kingdom\n",
        "\n",
        "# Сохраняем модифицированный DataFrame в новый файл\n",
        "dfr.to_csv(\"taxonomy_kingdom.csv\", index=False)\n",
        "\n",
        "# Выводим столбец Kingdom\n",
        "print(dfr[\"Kingdom\"])\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "id": "aNU_1UOiPFNH"
      },
      "outputs": [],
      "source": [
        "# # Функция для удаления первого значения из списков в столбце 'IdList'\n",
        "# def remove_first_value(x):\n",
        "#     if isinstance(x, (list, ListElement)):\n",
        "#         if len(x) > 1:\n",
        "#             return x[1]  # Возвращаем второй элемент списка\n",
        "#         elif len(x) == 1:\n",
        "#             return x[0]  # Если только один элемент, возвращаем его\n",
        "#         else:\n",
        "#             return np.nan\n",
        "#     else:\n",
        "#         return x  # Возвращаем значение как есть, если оно не список"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "id": "UO4ocBiiIMXw"
      },
      "outputs": [],
      "source": [
        "# import time\n",
        "# from Bio import Entrez\n",
        "# from Bio.Entrez.Parser import ListElement\n",
        "# Entrez.email = \"iroslov.01@gmail.com\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {
        "id": "SXqW7yMQvKUg"
      },
      "outputs": [],
      "source": [
        "# dfr['IdList'] = dfr['IdList'].apply(lambda x: x if isinstance(x, str) else str(x))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "id": "it9UXL1llan1"
      },
      "outputs": [],
      "source": [
        "# # Применяем функцию к столбцу 'IdList'\n",
        "# dfr['IdList'] = remove_first_value(dfr['IdList'])\n",
        "# dfr"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "id": "g1cg4N_Pl6J3"
      },
      "outputs": [],
      "source": [
        "# print(dfr['IdList'].apply(type).value_counts())"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "h9t8az4gKTQP"
      },
      "outputs": [
        {
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "Cell \u001b[1;32mIn[4], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# Создаем пустой список для хранения уровня таксономической классификации \"Kingdom\"\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m kingdom \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m      4\u001b[0m \u001b[38;5;66;03m# Создаем пустые DataFrame для бактерий и грибов\u001b[39;00m\n\u001b[0;32m      5\u001b[0m bacteria_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame()\n",
            "Cell \u001b[1;32mIn[4], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# Создаем пустой список для хранения уровня таксономической классификации \"Kingdom\"\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m kingdom \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m      4\u001b[0m \u001b[38;5;66;03m# Создаем пустые DataFrame для бактерий и грибов\u001b[39;00m\n\u001b[0;32m      5\u001b[0m bacteria_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame()\n",
            "File \u001b[1;32m_pydevd_bundle\\\\pydevd_cython.pyx:1698\u001b[0m, in \u001b[0;36m_pydevd_bundle.pydevd_cython.SafeCallWrapper.__call__\u001b[1;34m()\u001b[0m\n",
            "File \u001b[1;32m_pydevd_bundle\\\\pydevd_cython.pyx:636\u001b[0m, in \u001b[0;36m_pydevd_bundle.pydevd_cython.PyDBFrame.trace_dispatch\u001b[1;34m()\u001b[0m\n",
            "File \u001b[1;32m_pydevd_bundle\\\\pydevd_cython.pyx:1113\u001b[0m, in \u001b[0;36m_pydevd_bundle.pydevd_cython.PyDBFrame.trace_dispatch\u001b[1;34m()\u001b[0m\n",
            "File \u001b[1;32m_pydevd_bundle\\\\pydevd_cython.pyx:1091\u001b[0m, in \u001b[0;36m_pydevd_bundle.pydevd_cython.PyDBFrame.trace_dispatch\u001b[1;34m()\u001b[0m\n",
            "File \u001b[1;32m_pydevd_bundle\\\\pydevd_cython.pyx:496\u001b[0m, in \u001b[0;36m_pydevd_bundle.pydevd_cython.PyDBFrame.do_wait_suspend\u001b[1;34m()\u001b[0m\n",
            "File \u001b[1;32mc:\\Users\\Igorr\\Documents\\ITMO5grade\\Project_with_Susan\\Made_code\\Made_code\\Lib\\site-packages\\debugpy\\_vendored\\pydevd\\pydevd.py:2197\u001b[0m, in \u001b[0;36mPyDB.do_wait_suspend\u001b[1;34m(self, thread, frame, event, arg, exception_type)\u001b[0m\n\u001b[0;32m   2194\u001b[0m             from_this_thread\u001b[38;5;241m.\u001b[39mappend(frame_custom_thread_id)\n\u001b[0;32m   2196\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_threads_suspended_single_notification\u001b[38;5;241m.\u001b[39mnotify_thread_suspended(thread_id, thread, stop_reason):\n\u001b[1;32m-> 2197\u001b[0m         keep_suspended \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_do_wait_suspend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mthread\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mframe\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mevent\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43marg\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtrace_suspend_type\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfrom_this_thread\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mframes_tracker\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   2199\u001b[0m frames_list \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   2201\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m keep_suspended:\n\u001b[0;32m   2202\u001b[0m     \u001b[38;5;66;03m# This means that we should pause again after a set next statement.\u001b[39;00m\n",
            "File \u001b[1;32mc:\\Users\\Igorr\\Documents\\ITMO5grade\\Project_with_Susan\\Made_code\\Made_code\\Lib\\site-packages\\debugpy\\_vendored\\pydevd\\pydevd.py:2266\u001b[0m, in \u001b[0;36mPyDB._do_wait_suspend\u001b[1;34m(self, thread, frame, event, arg, trace_suspend_type, from_this_thread, frames_tracker)\u001b[0m\n\u001b[0;32m   2263\u001b[0m                 queue\u001b[38;5;241m.\u001b[39mput(internal_cmd)\n\u001b[0;32m   2264\u001b[0m                 wait_timeout \u001b[38;5;241m=\u001b[39m TIMEOUT_FAST\n\u001b[1;32m-> 2266\u001b[0m         \u001b[43mnotify_event\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwait_timeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   2267\u001b[0m         notify_event\u001b[38;5;241m.\u001b[39mclear()\n\u001b[0;32m   2269\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n",
            "File \u001b[1;32mC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\\Lib\\threading.py:629\u001b[0m, in \u001b[0;36mEvent.wait\u001b[1;34m(self, timeout)\u001b[0m\n\u001b[0;32m    627\u001b[0m signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_flag\n\u001b[0;32m    628\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m signaled:\n\u001b[1;32m--> 629\u001b[0m     signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cond\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    630\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m signaled\n",
            "File \u001b[1;32mC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\\Lib\\threading.py:331\u001b[0m, in \u001b[0;36mCondition.wait\u001b[1;34m(self, timeout)\u001b[0m\n\u001b[0;32m    329\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    330\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m--> 331\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    332\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    333\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m waiter\u001b[38;5;241m.\u001b[39macquire(\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
            "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ],
      "source": [
        "# Создаем пустой список для хранения уровня таксономической классификации \"Kingdom\"\n",
        "kingdom = []\n",
        "\n",
        "# Создаем пустые DataFrame для бактерий и грибов\n",
        "bacteria_df = pd.DataFrame()\n",
        "fungi_df = pd.DataFrame()\n",
        "\n",
        "# Итерация по DataFrame\n",
        "for i in range(len(dfr)):\n",
        "    id_value = dfr['IdList'].iloc[i]\n",
        "\n",
        "    # Проверяем, является ли ID допустимым\n",
        "    if pd.isna(id_value) or not isinstance(id_value, (str, int)):\n",
        "        print(f\"Некорректный ID на индексе {i}: {id_value}\")\n",
        "        kingdom.append(np.nan)\n",
        "        continue\n",
        "\n",
        "    id_value = str(id_value)  # Преобразуем ID в строку\n",
        "\n",
        "    try:\n",
        "        handle = Entrez.efetch(db=\"Taxonomy\", id=id_value, retmode=\"xml\")\n",
        "        records = Entrez.read(handle)\n",
        "        handle.close()\n",
        "\n",
        "        # Получаем таксономическую линию\n",
        "        tax = records[0][\"Lineage\"].split(\"; \")\n",
        "        time.sleep(0.2)\n",
        "\n",
        "        # Проверяем, являются ли это грибами или бактериями\n",
        "        if any(\"Fungi\" in tax_elem for tax_elem in tax):\n",
        "            # Добавляем уровень \"Kingdom\" в список\n",
        "            kingdom.append(tax[3] if len(tax) > 3 else np.nan)\n",
        "            # Добавляем данные в DataFrame грибов\n",
        "            fungi_df = pd.concat([fungi_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "        elif any(\"Bacteria\" in tax_elem for tax_elem in tax):\n",
        "            # Добавляем уровень \"Kingdom\" в список\n",
        "            kingdom.append(tax[1] if len(tax) > 1 else np.nan)\n",
        "            # Добавляем данные в DataFrame бактерий\n",
        "            bacteria_df = pd.concat([bacteria_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "        else:\n",
        "            kingdom.append(np.nan)\n",
        "    except Exception as e:\n",
        "        print(f\"Ошибка при обработке ID {id_value} на индексе {i}: {e}\")\n",
        "        kingdom.append(np.nan)\n",
        "\n",
        "# Создаем новый столбец в DataFrame для уровня классификации \"Kingdom\"\n",
        "dfr[\"Kingdom\"] = kingdom\n",
        "\n",
        "# Сохраняем модифицированный DataFrame в новый файл\n",
        "dfr.to_csv(\"taxonomy_kingdom.csv\", index=False)\n",
        "\n",
        "# Выводим столбец Kingdom\n",
        "print(dfr[\"Kingdom\"])"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "11XhnvT9mRiF"
      },
      "source": [
        "##List convert to str"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Lhqi5JMt-au8"
      },
      "outputs": [],
      "source": [
        "# dfr['IdList'] = dfr['IdList'].apply(lambda x: x if isinstance(x, str) else str(x))\n",
        "# # dfr = dfr.drop([4,8])\n",
        "# dfr"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5M4AqMmCB9CP"
      },
      "outputs": [],
      "source": [
        "# dfr.reset_index(drop=True, inplace=True)\n",
        "# dfr"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "-mWAQXVXXI7J"
      },
      "outputs": [],
      "source": [
        "# # Создаем пустые списки для хранения каждого уровня таксономической классификации\n",
        "# superkingdom = []\n",
        "# clade = []\n",
        "# kingdom = []\n",
        "# phylum = []\n",
        "# class_ = []\n",
        "# order = []\n",
        "# family = []\n",
        "# genus = []\n",
        "# species = []\n",
        "\n",
        "# # Создаем пустые DataFrame для бактерий и грибов\n",
        "# bacteria_df = pd.DataFrame()\n",
        "# fungi_df = pd.DataFrame()\n",
        "\n",
        "# # Итерация по DataFrame\n",
        "# for i in range(len(dfr)):\n",
        "#     id_value = dfr['IdList'][i]\n",
        "\n",
        "#     # Проверяем, является ли ID допустимым\n",
        "#     if pd.isna(id_value) or not isinstance(id_value, str):\n",
        "#         print(f\"Некорректный ID на индексе {i}: {id_value}\")\n",
        "#         # Добавляем np.nan для каждого уровня классификации\n",
        "#         superkingdom.append(np.nan)\n",
        "#         kingdom.append(np.nan)\n",
        "#         clade.append(np.nan)\n",
        "#         phylum.append(np.nan)\n",
        "#         class_.append(np.nan)\n",
        "#         order.append(np.nan)\n",
        "#         family.append(np.nan)\n",
        "#         genus.append(np.nan)\n",
        "#         species.append(np.nan)\n",
        "#         continue\n",
        "\n",
        "#     try:\n",
        "#         handle = Entrez.efetch(db=\"Taxonomy\", id=id_value, retmode=\"xml\")\n",
        "#         records = Entrez.read(handle)\n",
        "\n",
        "#         # Получаем таксономическую линию\n",
        "#         tax = records[0][\"Lineage\"].split(\"; \")\n",
        "#         time.sleep(0.2)\n",
        "\n",
        "#         # Проверяем, являются ли это грибами или бактериями\n",
        "#         if \"Fungi\" in tax:\n",
        "#             # Извлекаем каждый уровень классификации и добавляем в соответствующий список\n",
        "#             superkingdom.append(tax[1] if len(tax) > 1 else np.nan)\n",
        "#             kingdom.append(tax[3] if len(tax) > 3 else np.nan)\n",
        "#             clade.append(tax[2] if len(tax) > 2 else np.nan)\n",
        "#             phylum.append(tax[4] if len(tax) > 4 else np.nan)\n",
        "#             class_.append(tax[5] if len(tax) > 5 else np.nan)\n",
        "#             order.append(tax[6] if len(tax) > 6 else np.nan)\n",
        "#             family.append(tax[7] if len(tax) > 7 else np.nan)\n",
        "#             genus.append(tax[8] if len(tax) > 8 else np.nan)\n",
        "#             species.append(tax[-1] if len(tax) > 0 else np.nan)\n",
        "#             # Добавляем данные в DataFrame грибов\n",
        "#             fungi_df = pd.concat([fungi_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "\n",
        "#         elif \"Bacteria\" in tax:\n",
        "#             # Извлекаем каждый уровень классификации и добавляем в соответствующий список\n",
        "#             kingdom.append(tax[1] if len(tax) > 1 else np.nan)\n",
        "#             if \"group\" in tax[2]:\n",
        "#                 clade.append(tax[2] if len(tax) > 2 else np.nan)\n",
        "#                 phylum.append(tax[3] if len(tax) > 3 else np.nan)\n",
        "#                 class_.append(tax[4] if len(tax) > 4 else np.nan)\n",
        "#                 order.append(tax[5] if len(tax) > 5 else np.nan)\n",
        "#                 family.append(tax[6] if len(tax) > 6 else np.nan)\n",
        "#                 genus.append(tax[7] if len(tax) > 7 else np.nan)\n",
        "#                 species.append(tax[-1] if len(tax) > 0 else np.nan)\n",
        "#             else:\n",
        "#                 clade.append(np.nan)\n",
        "#                 phylum.append(tax[2] if len(tax) > 2 else np.nan)\n",
        "#                 class_.append(tax[3] if len(tax) > 3 else np.nan)\n",
        "#                 order.append(tax[4] if len(tax) > 4 else np.nan)\n",
        "#                 family.append(tax[5] if len(tax) > 5 else np.nan)\n",
        "#                 genus.append(tax[6] if len(tax) > 6 else np.nan)\n",
        "#                 species.append(tax[-1] if len(tax) > 0 else np.nan)\n",
        "#             # Добавляем данные в DataFrame бактерий\n",
        "#             bacteria_df = pd.concat([bacteria_df, dfr.iloc[[i]]], ignore_index=True)\n",
        "\n",
        "#     except Exception as e:\n",
        "#         print(f\"Ошибка при обработке ID {id_value} на индексе {i}: {e}\")\n",
        "#         # Добавляем np.nan для каждого уровня классификации\n",
        "#         superkingdom.append(np.nan)\n",
        "#         kingdom.append(np.nan)\n",
        "#         clade.append(np.nan)\n",
        "#         phylum.append(np.nan)\n",
        "#         class_.append(np.nan)\n",
        "#         order.append(np.nan)\n",
        "#         family.append(np.nan)\n",
        "#         genus.append(np.nan)\n",
        "#         species.append(np.nan)\n",
        "\n",
        "# # Создаем новые столбцы в DataFrame для каждого уровня классификации\n",
        "# dfr[\"Kingdom\"] = kingdom\n",
        "# dfr[\"clade\"] = clade\n",
        "# dfr[\"Phylum\"] = phylum\n",
        "# dfr[\"Class\"] = class_\n",
        "# dfr[\"Order\"] = order\n",
        "# dfr[\"Family\"] = family\n",
        "# dfr[\"Genus\"] = genus\n",
        "# dfr[\"Species\"] = species\n",
        "\n",
        "# # Сохраняем модифицированный DataFrame в новый файл\n",
        "# dfr.to_csv(\"taxonomy_descriptor.csv\", index=False)\n",
        "\n",
        "\n",
        "# print(Kingdom)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "6upJ0ZLiEJpH"
      },
      "outputs": [],
      "source": [
        "dfr\n",
        "#The information is not true so the information is obtained again"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "zePjGVwufS47"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Список единиц измерения\n",
        "concentration_units = [\n",
        "    'uM', 'mm', 'mg/dL', 'mg/L', 'IU/L', 'g', 'u/L', 'hr', 'CELLS/uL', 'g/dL',\n",
        "    'meq/L', 'G/DL', 'MEQ/L', 'mg kg-1', 's', 'ug/L', 'mM', 'umol/L', '/s',\n",
        "    'mg/ml', 'x10E3/uL', 'nM', 'pg', 'fL', 'x10_2/uL', 'x10_4/uL', 'x10_4/ul',\n",
        "    '10\\'-2 umol/ml', 'umol/ml', 'PG', 'FL', 'x10E6/uL', 'MMOL/L', 'mg l-1',\n",
        "    '/100WBC', 'cm', '/uM/s', '/s/microM', '/hr', 'um', 'ng/ml', 'M',\n",
        "    'log10CFU/ml', 'mm2', 'mg ml-1', '/ml', 'log10CFU', 'ug', 'uM/ml',\n",
        "    '10\\'-2microM', '/mM/s', 'min', 'microM', 'millimeter', 'uM/L', 'day',\n",
        "    'mm3', 'mmHg', 'microg/ml', '10\\'8/ml', 'uM l-1', 'mg.hr/L', 'ug mg-1',\n",
        "    '10\\'-2micromol/ml', 'mmol/L', 'nm', 'ml/min.kg', '10\\'5/ml', '10\\'6/M/s',\n",
        "    'U', 'microg.hr/mL', '10\\'5/M/s', 'mg Kg-1', '10\\'-3micromol/ml', 'log10CFU/ml.hr',\n",
        "    'mg/kg/day', 'ug.hr/ml', 'nmol/ml.min', 'g ml-1', 'deltalog10CFU',\n",
        "    '/M/s', 'mg mouse-1', '10\\'3CFU/ml', 'mmol/ml', '10^4/ml', 'nmol/ml',\n",
        "    'uL/ml', 'CFU', 'U/min', '10\\'-2umol', 'log10CFU/g', '10^3/ml', 'umol',\n",
        "    'ng', '10^8/ml', 'U/L', '10\\'4/M/s', '10^9/ml', '10\\'-2mmol/ml',\n",
        "    'ug/uL', 'L/Kg', 'g/ml', 'ug hr ml-1', 'nmol/min/mg', 'Kg', 'mg',\n",
        "    '10^2/ml', 'ppm', 'deltalog10CFU/ml', 'a.u.', 'ml min-1 kg-1', '10^7/ml',\n",
        "    '10^5/ml', '10^6/ml', 'pmol', 'hr.microg/ml', 'ml', 'pg/ml', 'uL/hr/cm2',\n",
        "    'mm/mg', 'CFU/ml', '10\\'-3microM', 'M min-1 (M of enzyme)-1', '10\\'7/M/s',\n",
        "    'nmol', 'ug/ml.hr', 'microg', 'ml/g', 'ug kg-1', 'ng ml-1', 'microg/cm3',\n",
        "    'ug ml-1 ml-1', 'uM/hr', '10\\'-3 log10CFU/ml/hr', 'ug mL-1', 'nmol/L',\n",
        "    'nmol/min', 'ml/kg', '10^-3mM', 'ug/ml/hr', '10\\'-3uM/ml', 'mg kg-1 day-1',\n",
        "    'ng g-1', 'ml/hr.kg', 'l kg-1', '10\\'3/M/s', '10\\'-7No_unit', 'uM min-1mg-1',\n",
        "    '10e4 M-1 s-1', 'U/ml', 'ug min ml-1', 'M l-1', '10\\'-3 ug/ml', 'L/hr/kg',\n",
        "    'microL/min/mg', 'mg.min/L', '10\\'-6 ug/ml', '10\\'-8M', '10\\'3/M',\n",
        "    'deltalog10CFU/ml.hr', '10\\'-7cm2/sec', 'log10RLU', '10^7/g', 'cpm/min',\n",
        "    'pmol/min', '10\\'-6M', 'Ratio', 'microg.min/mL', '10^-8M', '10^-9No_unit',\n",
        "    'uM kg-1', 'mmHg/min', 'um/s', 'nmol/sec', '10^-1/s', '10\\'2/M/s',\n",
        "    '10^-6/s', '10^1/Ms', '(g kidney)-1', 'ng hr ml-1', 'microg/min/ml',\n",
        "    '10\\'6/ml', '%ID', 'N', '10^-1M', '10^8/g', '10\\'6/s/M', '10\\'10CFU',\n",
        "    'ug/L/h', 'hr.ng/ml', 'ng.hr/ml', 'g/dl', 'log CFU', '10\\'-5 ug/ml',\n",
        "    'umol/min/mg', 'ug/sq.cm', '10\\'7/g', 'nm/s', '10\\'9/L', '10\\'9/g',\n",
        "    's-1', '10\\'8CFU/ml', '1/Ms', 'ucm/s', '10\\'2 ug/ml', '10\\'-6cm2/s',\n",
        "    '10\\'-2 ug/ml', 'microg/cm2', '10\\'-1/hr', 'uM min-1', 'umol/Kg',\n",
        "    'ug min-1 ml-1', '10\\'4CFU/ml', 'RLU', '10\\'3L/mol', 'l', '10\\'7/ml',\n",
        "    '10\\'2/ml', '10\\'3/ml', 'ml/min.Kg', '10\\'2 uM/s', '10\\'5/s/M',\n",
        "    '10\\'6CFU/ml', '10\\'7CFU/ml', '10\\'2/g', 'cm s-1', 'M-1', '10^-5No_unit',\n",
        "    '10^6cm/s', '1/M*s', '1/M', '10\\'9CFU', 'se-1', '/uL', '10\\'8CFU',\n",
        "    '10\\'5CFU', 'g/L', 'L/kg', 'mmol/min/mg', 'ml hr-1 kg-1', 'ml kg-1',\n",
        "    '10\\'12/L', 'ug/g', 'uM.hr/ml', '10\\'7CFU', '10\\'-6 mol/L', '10^7CFU',\n",
        "    '10^7CFU/g', '10^2/Ms', 'mg kg-1 p.o.', '1/S', 'weeks', '10\\'-4 ug/ml',\n",
        "    'mg/kg', 'ug ml-1 hr-1', '10\\'-7 ug/ml', 'ug.hr/L'\n",
        "]\n",
        "\n",
        "# Функция для перевода величин в стандартный вид [ug.mL-1]\n",
        "def convert_to_ug_mL(concentration):\n",
        "    conversion_factors = {\n",
        "        'uM': 1e-6,\n",
        "        'mm': 1e-3,\n",
        "        'mg/dL': 1e-3,\n",
        "        'mg/L': 1e-6,\n",
        "        'IU/L': 1e-6,\n",
        "        'g': 1e-6,\n",
        "        'u/L': 1e-6,\n",
        "        'CELLS/uL': 1e-6,\n",
        "        'g/dL': 1e-6,\n",
        "        'meq/L': 1e-6,\n",
        "        'ug/L': 1e-6,\n",
        "        'mM': 1e-3,\n",
        "        'umol/L': 1e-6,\n",
        "        'nM': 1e-9,\n",
        "        'pg': 1e-12,\n",
        "        'mg/ml': 1e-3,\n",
        "        'ug': 1e-6,\n",
        "        'uM/ml': 1e-6,\n",
        "        'mmol/L': 1e-3,\n",
        "        'mg/kg': 1e-3,\n",
        "        'microM': 1e-6,\n",
        "        'nmol': 1e-9,\n",
        "        'mg': 1e-3,\n",
        "        'g/ml': 1e-3,\n",
        "        'ug/mL': 1e-6,\n",
        "    }\n",
        "\n",
        "    if concentration in conversion_factors:\n",
        "        # Конвертируем в ug.mL-1\n",
        "        return 1 / conversion_factors[concentration]  # возвращаем коэффициент, чтобы выразить в ug.mL-1\n",
        "    return None\n",
        "\n",
        "# Конвертация всех величин в ug.mL-1\n",
        "standard_values = {unit: convert_to_ug_mL(unit) for unit in concentration_units}\n",
        "standard_values = {k: v for k, v in standard_values.items() if v is not None}\n",
        "\n",
        "# Фильтрация неконцентрированных величин\n",
        "non_concentration_units = [\n",
        "    'hr', 's', '/s', 'cm', '/100WBC', 'day', 'fl', 'min', 'g', 'mL',\n",
        "    'uL/hr/cm2', 'CFU', 'U/min', 'log10CFU', 'pg/ml', 'uL/ml', 'ml',\n",
        "    'g/dL', 'log10CFU/g', '10\\'-5 ug/ml', 'weeks'\n",
        "]\n",
        "\n",
        "# Вывод результатов\n",
        "print(\"Концентрации в ug.mL-1:\")\n",
        "for original, standard in standard_values.items():\n",
        "    print(f\"{original}: {standard:.2f} ug.mL-1\")\n",
        "\n",
        "print(\"\\nНеконцентрированные величины:\")\n",
        "for non_concentration in non_concentration_units:\n",
        "    print(non_concentration)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jLxoFpWQjMpQ"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Made_code",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.13.0"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
